rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -0,0 +1,365 @@
+        today_epoch = int(datetime.combine(datetime.today(), time.min).timestamp())
+        param_types = {
+            '$limit': 'Int!',
+            '$offset': 'Int!',
+            '$token_ids': '[String!]',
+            '$datetime': 'Int!',
+        }
+        param_values = {
+            'limit': limit,
+            'offset': 0,
+            'token_ids': unknown_assets_addresses_lower,
+            'datetime': today_epoch,
+        }
+        while True:
+            result = self.graph.query(  # type: ignore
+                querystr=querystr,
+                param_types=param_types,
+                param_values=param_values,
+            )
+            result_data = result['tokenDayDatas']
+
+            if not result_data:
+                break
+
+            for tdd in result_data:
+                asset_price[to_checksum_address(tdd['token']['id'])] = FVal(tdd['priceUSD'])
+
+            # Update pagination step
+            param_values['offset'] += limit  # type: ignore
+
+        return asset_price
+
+    def _update_assets_prices_in_address_balances(
+            self,
+            address_balances: AddressBalances,
+            known_asset_price: AssetPrice,
+            unknown_asset_price: AssetPrice,
+    ) -> None:
+        """Update the pools underlying assets prices in USD (prices obtained
+        via Inquirer and the Uniswap subgraph)
+        """
+        for lps in address_balances.values():
+            for lp in lps:
+                # Try to get price from either known or unknown asset price.
+                # Otherwise keep existing price (zero)
+                total_user_balance = FVal(0)
+                for asset in lp.assets:
+                    asset_ethereum_address = asset.asset.ethereum_address
+                    asset_usd = known_asset_price.get(
+                        asset_ethereum_address,
+                        unknown_asset_price.get(asset_ethereum_address, ZERO),
+                    )
+                    # Update <LiquidityPoolAsset> if asset USD price exists
+                    if asset_usd != ZERO:
+                        asset.asset_usd = Price(asset_usd)
+                        asset.user_balance.usd_value = (
+                            asset.user_balance.amount * asset_usd
+                        )
+
+                    total_user_balance += asset.user_balance.usd_value
+
+                # Update <LiquidityPool> total balance in USD
+                lp.user_balance.usd_value = total_user_balance
+
+    def get_balances(
+        self,
+        addresses: List[ChecksumEthAddress],
+        graph_query: bool,
+    ) -> AddressBalances:
+        """Get the addresses' balances in the Uniswap protocol
+
+        Premium users can request balances either via the Uniswap subgraph or
+        Zerion SDK.
+        """
+        is_graph_mode = self.graph and self.premium and graph_query
+
+        if is_graph_mode:
+            protocol_balance = self._get_balances_graph(addresses)
```

Comment:
nitpick: `protocol_balances`

Reply:
I've kept it singular as it is an instance of `ProtocolBalance`, and pluralised `<ProtocolBalance>.address_balances` (a dict of address - LP balances). Naming drives me crazy tbh, I try to do my best but there are times where I can't right pick singular or plural lol.